Readme.md

```diff
@@ -1,5 +1,298 @@
 # Flux
 
-lexer generator for any context free syntax in 100% rust
+A lexer generator for any context-free syntax in 100% rust with no dependencies.
 
-created for [Fender-lang](https://github.com/FenderLang/)
+Created for [Fender-lang](https://github.com/FenderLang/).
+
+# Lexer
+
+## What is a Lexer
+- A Lexer is part of an interpreter that takes sequence of characters and turns them into tokens.
+
+## How to use the Lexer
+
+First define your BNF input
+```rust
+let bnf_input = include_str!("path/to/example.bnf");
+```
+
+To parse the input you should use `flux_bnf::bnf::parse`
+```rust
+let mut lexer = flux_bnf::bnf::parse(bnf_input).unwrap();     
+```
+
+## How to set Lexer Rules
+Now that you have `Lexer` we can use that to now set some rules
+
+Before being able to set some rules you will need write a BNF describing the syntax which you can find [here](#bnf)
```

Comment:
Might be worth moving the link to the BNF subsection up to the part where you mention defining BNF input, so they have a link explaining what BNF is the first time it's mentioned instead of the second.